objects/oplayercontroller/Step_0.gml

```diff
@@ -558,6 +618,76 @@ if (!isGrabbed)
 // Handle Enviornmental Displacement
 environmentDisplacement = 0;
 
+// Handle freezing screen
+if (state == eState.SCREEN_FREEZE)
+{
+	image_speed = 0;
+	hsp = 0;
+	environmentDisplacement = 0;
+	vsp = 0;

```

Comment:
Instead of storing the speed values for the player, you might be able to get away with just not updating the player's movement during screen freeze.
The lines to update player movement are 2137-2139 in oPlayerController's Step event, you can simply add a condition there the see if the player is in screen freeze, although this would involve making the same check twice in the code, making it more messy. This works fine for now, but it's something to consider in the future.